lib/spack/llnl/util/symlink.py

```diff
@@ -42,21 +42,23 @@ def symlink(real_path, link_path):
 
 
 def islink(path):
-    return os.path.islink(path) or _win32_is_junction(path)
+    return Path(path).is_link() or _win32_is_junction(path)
 
 
 # '_win32' functions based on
 # https://github.com/Erotemic/ubelt/blob/master/ubelt/util_links.py
 def _win32_junction(path, link):
+    path = Path(path)
+    link = Path(link)
     # junctions require absolute paths
-    if not os.path.isabs(link):
-        link = os.path.abspath(link)
+    if not link.is_absolute():
+        link = link.absolute()
 
     # os.symlink will fail if link exists, emulate the behavior here
-    if exists(link):
-        raise OSError(errno.EEXIST, "File  exists: %s -> %s" % (link, path))
+    if link.exists():
+        raise OSError(errno.EEXIST, "File  exists: %s -> %s" % (str(link), str(path)))
 
-    if not os.path.isabs(path):
+    if not path.is_absolute():
         parent = os.path.join(link, os.pardir)
         path = os.path.join(parent, path)
         path = os.path.abspath(path)
```

Comment:
Why not also use path operations here? I suppose because of the regex?

Reply:
Yeah, this was just an oversight.